$MirOS: ports/www/php/extensions/patches/patch-ext_gd_config_m4,v 1.1.7.1 2005/03/18 16:00:00 tg Exp $
$OpenBSD: patch-ext_gd_config_m4,v 1.1.1.1 2004/10/02 11:32:35 robert Exp $
--- ext/gd/config.m4.orig	Tue Jan  4 06:51:36 2005
+++ ext/gd/config.m4	Wed May  4 17:52:19 2005
@@ -87,13 +87,18 @@ AC_DEFUN([PHP_GD_PNG],[
       AC_MSG_ERROR([PNG support requires ZLIB. Use --with-zlib-dir=<DIR>])
     fi
 
-    if test ! -f $GD_PNG_DIR/include/png.h; then
+    if test ! -f $GD_PNG_DIR/include/png.h && test ! -f $GD_PNG_DIR/include/libpng/png.h; then
       AC_MSG_ERROR([png.h not found.])
     fi
 
     PHP_CHECK_LIBRARY(png,png_write_image,
     [
+     if test -f $GD_PNG_DIR/include/png.h; then
       PHP_ADD_INCLUDE($GD_PNG_DIR/include)
+     fi
+     if test -f $GD_PNG_DIR/include/libpng/png.h; then
+      PHP_ADD_INCLUDE($GD_PNG_DIR/include/libpng)
+     fi
       PHP_ADD_LIBRARY_WITH_PATH(z, $PHP_ZLIB_DIR/lib, GD_SHARED_LIBADD)
       PHP_ADD_LIBRARY_WITH_PATH(png, $GD_PNG_DIR/lib, GD_SHARED_LIBADD)
     ],[
